### PR TITLE
Add publish-to-bcr configuration

### PR DESCRIPTION
**Pull request by jsharpe**
Adds configuration for the publish-to-bcr app which auto publishes a PR to the bazel-central-registry whenever a release is made.

### CODE REVIEW

File: .bcr/config.yml
Comment (jsharpe) [inline .bcr/config.yml:3]: We may be able to remove this if bazel-contrib/bazel-central-registry gets the publish-to-bcr app configured on it.

File: .bcr/presubmit.yml
Comment (jsharpe) [inline .bcr/presubmit.yml:3]: We can't run tests in the registry until we have a hermetic toolchain as there will be no cuda install in bazelbuild CI

File: .bcr/config.yml
Comment (jsharpe) [reply .bcr/config.yml]: App should be configured for all repos in bazel-contrib so this should work ok without a fixed releaser.

File: .bcr/metadata.template.json
Comment (jsharpe) [inline .bcr/metadata.template.json:8]: @cloudhan @ryanleary note that this is the maintainer of the BCR entry not necessarily the repo; feel free to add yourself to this list if you want but I'm happy to be listed here.

**Comment by jsharpe** [comment]
Once this is merged we can try cutting a release and it should post a PR to the central registry to add rules_cuda as a module.

**Review by cloudhan — APPROVED** [review]

**Pull request merged**
